thebeetoken/beenest-web
# Use BTC wallet address from backend

## Description
See thebeetoken/beenest-backend#676

## How to Test
This is a pure refactor, from the front-end perspective, so just regression test

(For bonus points, change `btcWalletAddress` on a booking in the database and verify that it displays client-side on the receipt page.)

Which devices did you test on?

## REVIEWERS:
Check against these principles:

### High level
Does this code need to be written?
What are the alternatives?
Will this implementation become a support issue?
How much error margin does this solution have?

### Code
* Does the code follow industry standards?
JS: https://github.com/airbnb/javascript
React: https://github.com/airbnb/javascript/tree/master/react
https://github.com/vasanthk/react-bits
Documentation headers: http://usejsdoc.org/index.html

* Is there duplicated code? Can it be refactored into a shared method?
* Is the code consistent with our project?
* Are there unit tests? Do they test the states?
* Is the person refactoring another developer's code? If possible, did the original developer approve?

### Variables/Naming:
* Would the variable type led to future edge cases?
* Are the variable naming clear? Would the value contain something other than what the name describes.

### Security
* Can this be hacked or abused by the user?

## Further Work
If BTC payments get any use, we'll probably want to start generating new payment addresses per-booking on the backend

## Learnings
When you [procrastinate](https://github.com/thebeetoken/beenest-web/pull/79#commitcomment-32029958) on everything but the easy parts, it's all easy parts.
